assets/js/components/user-input/UserInputPreview.js

```diff
@@ -0,0 +1,128 @@
+	USER_INPUT_ANSWERS_HELP_NEEDED,
+	USER_INPUT_ANSWERS_POST_FREQUENCY,
+	USER_INPUT_ANSWERS_ROLE,
+} from './util/constants';
+const { useSelect, useDispatch } = Data;
+
+export default function UserInputPreview( { goTo } ) {
+	const settings = useSelect( ( select ) => select( CORE_USER ).getUserInputSettings() );
+	const {
+		hasStartedSavingInputSettings,
+		hasFinishedSavingInputSettings,
+	} = useSelect( ( select ) => ( {
+		hasStartedSavingInputSettings: select( CORE_USER ).hasStartedSavingInputSettings(),
+		hasFinishedSavingInputSettings: select( CORE_USER ).hasFinishedSavingInputSettings(),
+	} ) );
+
+	const { saveUserInputSettings } = useDispatch( CORE_USER );
+	const submitChanges = useCallback( () => {
+		saveUserInputSettings( settings );
+	}, [ settings ] );
+
+	return (
+		<div className="googlesitekit-user-input__preview">
+			<Row>
+				<Cell>
+					{ hasStartedSavingInputSettings && ! hasFinishedSavingInputSettings && (
+						<ProgressBar />
+					) }
+					{ ! hasStartedSavingInputSettings && (
+						<Fragment>
+							<Row>
+								<Cell lg={ 6 }>
+									<UserInputPreviewGroup
+										title={ __( '1 — Which best describes your team/role relation to this site?', 'google-site-kit' ) }
+										edit={ goTo.bind( null, 5 ) }
+										values={ settings?.role?.values || [] }
+										options={ USER_INPUT_ANSWERS_ROLE }
+									/>
+
+									<UserInputPreviewGroup
+										title={ __( '2 — How often do you create new posts for this site?', 'google-site-kit' ) }
+										edit={ goTo.bind( null, 4 ) }
+										values={ settings?.postFrequency?.values || [] }
+										options={ USER_INPUT_ANSWERS_POST_FREQUENCY }
+									/>
+
+									<UserInputPreviewGroup
+										title={ __( '3 — What are the goals of this site?', 'google-site-kit' ) }
+										edit={ goTo.bind( null, 3 ) }
+										values={ settings?.goals?.values || [] }
+										options={ USER_INPUT_ANSWERS_GOALS }
+									/>
+								</Cell>
+								<Cell lg={ 6 }>
+									<UserInputPreviewGroup
+										title={ __( '4 — What do you need help most with for this site?', 'google-site-kit' ) }
+										edit={ goTo.bind( null, 2 ) }
+										values={ settings?.helpNeeded?.values || [] }
+										options={ USER_INPUT_ANSWERS_HELP_NEEDED }
+									/>
+
+									<UserInputPreviewGroup
+										title={ __( '5 — To help us identify opportunities for your site, enter the top three search terms that you’d like to show up for:', 'google-site-kit' ) }
+										edit={ goTo.bind( null, 1 ) }
+										values={ settings?.searchTerms?.values || [] }
+									/>
+								</Cell>
+							</Row>
+
+							<div className="googlesitekit-user-input__buttons">
+								<UserInputQuestionNotice />
+								<div>
+									<Button text onClick={ goTo.bind( null, 1 ) }>{ __( 'Back', 'google-site-kit' ) }</Button>
+									<Button onClick={ submitChanges }>{ __( 'Submit', 'google-site-kit' ) }</Button>
+								</div>
+							</div>
```

Comment:
I think it would be great to add some error handling to this component, for the case that the API request to save unexpectedly fails. We can use the `core/user` store's `getErrorForAction( 'saveUserInputSettings', [] )` selector, and if there is one, display it via an `ErrorNotice`. Maybe close to the buttons would work well.

Reply:
Updated.